### PR TITLE
feat: oscal normalize major changes to support new OSCAL 1.0.0

### PR DESCRIPTION
**Pull request by fsuits**
## Types of changes

- \[ \] Bug fix (non-breaking change which fixes an issue)
- \[x] New feature (non-breaking change which adds functionality)
- \[x] Breaking change (fix or feature that would cause existing functionality to change)
- \[x] My code follows the code style of this project.
- \[ \] My change requires a change to the documentation.
- \[ \] I have updated the documentation accordingly.
- \[x] I have added tests to cover my changes.
- \[x] All new and existing tests passed.
- \[x] All commits are signed-off.

Significant changes in oscal models and tests to support OSCAL 1.0.0

### CODE REVIEW

File: tests/data/split_merge/step4_split_groups_array/catalogs/mycatalog/catalog/groups/00000__group/controls/00001__control.json [hunk 1]
```diff
@@ -217,7 +217,7 @@
                                 "value": "c."
                             }
                         ],
-                        "prose": "Require {{ ac-2_prm_1 }} for group and role membership;"
+                        "prose": "Require {{ insert: param, ac-2_prm_1 }} for group and role membership;"
```
Comment (butler54) [inline tests/data/split_merge/step4_split_groups_array/catalogs/mycatalog/catalog/groups/00000__group/controls/00001__control.json:220]: This is crazy - created an issue for it here: https://github.com/usnistgov/OSCAL/issues/971

No action on our part except for with the SSP code. Can you open an issue on it please?

File: trestle/core/commands/split.py [hunk 2]
```diff
@@ -37,14 +37,16 @@
 
 def split_is_too_fine(split_paths: str, model_obj: OscalBaseModel) -> bool:
```
Comment (butler54) [inline trestle/core/commands/split.py:38]: Ensure an issue exists to track fixing this.

File: trestle/core/commands/author/ssp.py [hunk 3]
```diff
@@ -354,12 +358,13 @@ def _get_implementations(self, control_file: pathlib.Path) -> List[ossp.Implemen
                 by_comp: ossp.ByComponent = gens.generate_sample_model(ossp.ByComponent)
                 by_comp.description = text
                 statement: ossp.Statement = gens.generate_sample_model(ossp.Statement)
-                part_label = section.split(' ')[1].replace('.', '')
-                statement_label = f'{control_id}_smt.{part_label}'
-                statement.by_components = {statement_label: by_comp}
+                # the following may be needed with the 1.0.0 changes
+                # part_label = section.split(' ')[1].replace('.', '')  # noqa: E800
+                # statement_label = f'{control_id}_smt.{part_label}'   # noqa: E800
+                statement.by_components = [by_comp]
```
Comment (butler54) [inline trestle/core/commands/author/ssp.py:364]: Add issue to track this - related to this NIST parameter substitution issue.

**Review by butler54 — CHANGES_REQUESTED** [review]
I'm going to have to trust that a lot of the test data is okay. Make sure that issues are created in line with the comments.
A few more suggestions for which we could open issues:
- Optimization of common vs not (I think there is still room to give).
- Use of a smaller catalog object to minimize the schema changes that come along.

File: tests/data/split_merge/step4_split_groups_array/catalogs/mycatalog/catalog/groups/00000__group/controls/00001__control.json (hunk 1 above)
Comment (fsuits) [reply tests/data/split_merge/step4_split_groups_array/catalogs/mycatalog/catalog/groups/00000__group/controls/00001__control.json:220]: done

File: trestle/core/commands/split.py (hunk 2 above)
Comment (fsuits) [reply trestle/core/commands/split.py:38]: done

File: trestle/core/commands/author/ssp.py (hunk 3 above)
Comment (fsuits) [reply trestle/core/commands/author/ssp.py:364]: done

**Review by butler54 — APPROVED** [review]
LGTM

**Pull request merged**